SchemaValidator/SchemaValidation.py

```diff
@@ -0,0 +1,218 @@
+import json
+import yaml
+import urllib
+import ast
+from boto3 import client
+s3 = client('s3')
+from lxml import etree
+from sys import exc_info
+from urllib import request
+
+
+def validate_handler(event, context):
+    # This is the AWS Lambda handler function.
+    try:
+        # Get the uploaded file.
+        bucket = event['Records'][0]['s3']['bucket']['name']
+        xmlKey = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'], encoding='utf-8')
+        xmlFileObject = s3.get_object(Bucket=bucket, Key=xmlKey)
+        xmlText = xmlFileObject["Body"].read().decode()
+
+        # Parse xml and determine the file type. It might be hml or haml or ?, For now, assume hml.
+        fileType = 'hml'
+
+        # Fetch the Schema Text
+        if (fileType == 'hml'):
+            schemaKey = 'schema/hml-1.0.1.xsd'
+        elif (fileType == 'haml'):
+            raise Exception('I have not yet implemented validation of haml files.')
+        else:
+            raise Exception('Unknown file type:' + str(fileType))
+
+        schemaFileObject = s3.get_object(Bucket=bucket, Key=schemaKey)
+        schemaText = schemaFileObject["Body"].read().decode()
+
+        # Perform the validation.
+        validationResults = validateAgainstSchema(schemaText=schemaText, xmlText=xmlText)
+        print('ValidationResults:' + str(validationResults))
+
+        # Request the management app to update the validation status for this file.
+        url = getUrl()
+        token=getToken(url=url)
+
+        setValidationStatus(uploadFileName=xmlKey, isValid=(validationResults=='Valid'), validationFeedback=validationResults, url=url, token=token)
+
+        # TODO: Trigger Subsequent Data Processing (Parse XML into database tables?)
```

Comment:
Though it would be possible here, I'd put that in a single function.

Reply:
Yeah, I imagine that once a file is flagged as valid it could trigger another function to parse the file and do something with the data. I agree, it goes in another function and we'll discuss it.